src/LogicalGraph.ts

```diff
@@ -515,7 +515,7 @@ export class LogicalGraph {
         newNode.setEmbedKey(null);
 
         // convert start of end nodes to data components
-        if (newNode.getCategory() === Eagle.Category.Start || newNode.getCategory() === Eagle.Category.End) {
+        if (newNode.getCategory() === Eagle.Category.Start) {
```

Comment:
It looks like the "Eagle.Category.End" case of the switch within this if-block becomes redundant here.

We can probably remove the switch too, and just always use code in the "Eagle.Category.Start" case.